### ISSUE TITLE
[XCUITest] Failed to create WDA session (The application under test with bundle id '' is not running, possibly crashed) on Xcode 12.3

### ISSUE THREAD

**Issue by yongjie6** [appium/appium#14951]
## The problem

On ios platform, after overwriting and installing the tested app, appium cannot start the tested application normally, and the tested application always crashes

## Environment

* Appium Version ： 1.20.0-beta.0
* MAC OS:10.15.7
* Xcode: 12.2
* Node.js version:14.13.1
* Npm package manager:
* Mobile platform/version under test: 13.4.1
* Real device or emulator/simulator: Real device 
* Appium CLI or Appium.app|exe: Appium CLI 

## Details

I encountered a problem. After I installed a tested application, I could start the appium driver normally. At that time, after I overwrite the tested application and install it with appium, the tested application always crashed. I have tried a lot Try to change the parameters, but still can't solve

## Link to Appium logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here.
Do _NOT_ paste your full Appium logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include Appium logs!

https://gist.github.com/yongjie6/847da48b09d8097c609489495884a7ef

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by KazuCocoa** [comment]
> with Xcode 12C33


Xcode 12.3 cannot build WDA properly.
https://github.com/appium/WebDriverAgent/issues/432

Current `appium@beta` has the fix.
It has a few breaking changes, so please check https://github.com/appium/appium/pull/14904/files as well.

**Comment by yongjie6** [comment]
> > with Xcode 12C33
> 
> Xcode 12.3 cannot build WDA properly.
> [appium/WebDriverAgent#432](https://github.com/appium/WebDriverAgent/issues/432)
> 
> Current `appium@beta` has the fix. (appium-webdriveragent module should be `2.32.0`)
> It has a few breaking changes, so please check https://github.com/appium/appium/pull/14904/files as well.


I tried this version and still have problems

**Comment by yongjie6** [comment]
> > with Xcode 12C33
> 
> Xcode 12.3 cannot build WDA properly.
> [appium/WebDriverAgent#432](https://github.com/appium/WebDriverAgent/issues/432)
> 
> Current `appium@beta` has the fix. (appium-webdriveragent module should be `2.32.0`)
> It has a few breaking changes, so please check https://github.com/appium/appium/pull/14904/files as well.

The problem is that after overwriting and installing a higher version of the app, the appium driver does not work properly

**Comment by KazuCocoa** [comment]
Ah, I saw the below error first in your log, so I thought it was a known WDA build error.

```
2020-12-16 07:35:07:155 [Xcode] 
2020-12-16 07:35:07:161 [Xcode] error: Building for iOS, but the embedded framework 'CocoaAsyncSocket.framework' was built for iOS + iOS Simulator. (in target 'WebDriverAgentRunner' from project 'WebDriverAgent')
2020-12-16 07:35:07:161 [Xcode] 
2020-12-16 07:35:07:161 [Xcode] error: Building for iOS, but the embedded framework 'YYCache.framework' was built for iOS + iOS Simulator. (in target 'WebDriverAgentRunner' from project 'WebDriverAgent')
2020-12-16 07:35:07:161 [Xcode] 
2020-12-16 07:35:07:161 [Xcode] error: Building for iOS, but the linked and embedded framework 'CocoaAsyncSocket.framework' was built for iOS + iOS Simulator. (in target 'WebDriverAgentLib' from project 'WebDriverAgent')
2020-12-16 07:35:07:162 [Xcode] error: Building for iOS, but the linked and embedded framework 'YYCache.framework' was built for iOS + iOS Simulator. (in target 'WebDriverAgentLib' from project 'WebDriverAgent')
2020-12-16 07:35:07:162 [Xcode] 
2020-12-16 07:35:07:163 [Xcode] 
2020-12-16 07:35:07:163 [Xcode] 
2020-12-16 07:35:07:164 [Xcode] ** TEST BUILD FAILED **
2020-12-16 07:35:07:164 [Xcode] 
```

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-ios-device/blob/b134c0312657f4e30318796ab412c748c23bb686/lib/installation-proxy/index.js#L13


WDA installs app via `usbmuxd`. If it had something issue, a fresh installation also should have the same issue.
If it had an issue only when overriding the existing app, then probably the app under test has something issue when XCTest launches the app. (Appium internally calls XCTest framework's launch app API) Crash report may help then?


```
2020-12-16 07:35:10:169 [Xcode]     t =     0.20s Open com.yaymedialabs.putong.enterprise
2020-12-16 07:35:10:170 [Xcode] 
2020-12-16 07:35:10:170 [Xcode]     t =     0.20s     Launch com.yaymedialabs.putong.enterprise
2020-12-16 07:35:10:170 [Xcode] 
2020-12-16 07:35:10:313 [Xcode]     t =     0.34s         Wait for accessibility to load
2020-12-16 07:35:10:313 [Xcode] 
2020-12-16 07:35:10:324 [Xcode]     t =     0.35s         Setting up automation session
2020-12-16 07:35:10:324 [Xcode] 
2020-12-16 07:35:12:861 [Xcode]     t =     2.89s         Wait for com.yaymedialabs.putong.enterprise to idle
2020-12-16 07:35:12:862 [Xcode] 
2020-12-16 07:35:27:868 [Xcode] 2020-12-16 15:35:27.865968+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe it is idling
2020-12-16 07:35:27:868 [Xcode] 
2020-12-16 07:35:27:869 [Xcode] 2020-12-16 15:35:27.866572+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe there are no animations
2020-12-16 07:35:27:869 [Xcode] 
2020-12-16 07:35:31:359 [Xcode] 2020-12-16 15:35:31.358075+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:359 [Xcode] 
2020-12-16 07:35:31:478 [Xcode] 2020-12-16 15:35:31.477989+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:478 [Xcode] 
2020-12-16 07:35:31:595 [Xcode] 2020-12-16 15:35:31.595220+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:595 [Xcode] 
2020-12-16 07:35:31:713 [Xcode] 2020-12-16 15:35:31.712176+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:713 [Xcode] 
2020-12-16 07:35:31:829 [Xcode] 2020-12-16 15:35:31.828978+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:829 [Xcode] 
2020-12-16 07:35:31:942 [Xcode] 2020-12-16 15:35:31.942004+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:31:942 [Xcode] 
2020-12-16 07:35:32:056 [Xcode] 2020-12-16 15:35:32.055594+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:056 [Xcode] 
2020-12-16 07:35:32:169 [Xcode] 2020-12-16 15:35:32.168633+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:169 [Xcode] 
2020-12-16 07:35:32:284 [Xcode] 2020-12-16 15:35:32.282896+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:284 [Xcode] 
2020-12-16 07:35:32:402 [Xcode] 2020-12-16 15:35:32.402918+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:402 [Xcode] 
2020-12-16 07:35:32:515 [Xcode] 2020-12-16 15:35:32.514481+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:515 [Xcode] 
2020-12-16 07:35:32:628 [Xcode] 2020-12-16 15:35:32.627800+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:629 [Xcode] 
2020-12-16 07:35:32:745 [Xcode] 2020-12-16 15:35:32.744792+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:745 [Xcode] 
2020-12-16 07:35:32:860 [Xcode] 2020-12-16 15:35:32.859987+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:860 [Xcode] 
2020-12-16 07:35:32:975 [Xcode] 2020-12-16 15:35:32.974993+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:32:975 [Xcode] 
2020-12-16 07:35:33:092 [Xcode] 2020-12-16 15:35:33.092154+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:092 [Xcode] 
2020-12-16 07:35:33:205 [Xcode] 2020-12-16 15:35:33.205154+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:206 [Xcode] 
2020-12-16 07:35:33:322 [Xcode] 2020-12-16 15:35:33.322046+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:322 [Xcode] 
2020-12-16 07:35:33:440 [Xcode] 2020-12-16 15:35:33.439420+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:440 [Xcode] 
2020-12-16 07:35:33:555 [Xcode] 2020-12-16 15:35:33.554751+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:556 [Xcode] 
2020-12-16 07:35:33:675 [Xcode] 2020-12-16 15:35:33.675352+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:676 [Xcode] 
2020-12-16 07:35:33:788 [Xcode] 2020-12-16 15:35:33.787587+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:788 [Xcode] 
2020-12-16 07:35:33:905 [Xcode] 2020-12-16 15:35:33.905202+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:33:905 [Xcode] 
2020-12-16 07:35:34:019 [Xcode] 2020-12-16 15:35:34.019330+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:019 [Xcode] 
2020-12-16 07:35:34:133 [Xcode] 2020-12-16 15:35:34.133206+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:133 [Xcode] 
2020-12-16 07:35:34:246 [Xcode] 2020-12-16 15:35:34.246163+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:246 [Xcode] 
2020-12-16 07:35:34:362 [Xcode] 2020-12-16 15:35:34.362724+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:363 [Xcode] 
2020-12-16 07:35:34:479 [Xcode] 2020-12-16 15:35:34.479181+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:480 [Xcode] 
2020-12-16 07:35:34:598 [Xcode] 2020-12-16 15:35:34.598504+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:599 [Xcode] 
2020-12-16 07:35:34:715 [Xcode] 2020-12-16 15:35:34.715163+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:716 [Xcode] 
2020-12-16 07:35:34:832 [Xcode] 2020-12-16 15:35:34.832025+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:833 [Xcode] 
2020-12-16 07:35:34:949 [Xcode] 2020-12-16 15:35:34.948522+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:34:949 [Xcode] 
2020-12-16 07:35:35:063 [Xcode] 2020-12-16 15:35:35.062174+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:063 [Xcode] 
2020-12-16 07:35:35:176 [Xcode] 2020-12-16 15:35:35.176004+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:176 [Xcode] 
2020-12-16 07:35:35:290 [Xcode] 2020-12-16 15:35:35.288929+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:291 [Xcode] 
2020-12-16 07:35:35:409 [Xcode] 2020-12-16 15:35:35.409193+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:410 [Xcode] 
2020-12-16 07:35:35:522 [Xcode] 2020-12-16 15:35:35.521770+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:522 [Xcode] 
2020-12-16 07:35:35:640 [Xcode] 2020-12-16 15:35:35.639978+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:641 [Xcode] 
2020-12-16 07:35:35:757 [Xcode] 2020-12-16 15:35:35.755963+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:757 [Xcode] 
2020-12-16 07:35:35:879 [Xcode] 2020-12-16 15:35:35.877407+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:879 [Xcode] 
2020-12-16 07:35:35:996 [Xcode] 2020-12-16 15:35:35.995505+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:35:996 [Xcode] 
2020-12-16 07:35:36:109 [Xcode] 2020-12-16 15:35:36.109119+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:36:109 [Xcode] 
2020-12-16 07:35:36:226 [Xcode] 2020-12-16 15:35:36.225885+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:36:227 [Xcode] 
2020-12-16 07:35:36:348 [Xcode] 2020-12-16 15:35:36.346854+0800 WebDriverAgentRunner-Runner[735:122507] Using general test manager
2020-12-16 07:35:36:348 [Xcode] 
2020-12-16 07:35:36:461 [Xcode] 2020-12-16 15:35:36.460865+0800 WebDriverAgentRunner-Runner[735:122507] The application 'com.yaymedialabs.putong.enterprise' is not running in foreground after 5.00 seconds
2020-12-16 07:35:36:462 [Xcode] 
2020-12-16 07:35:36:471 [Xcode] 2020-12-16 15:35:36.4
```

**Comment by mykola-mokhnach** [comment]
> 12-16 07:35:12:861 [Xcode]     t =     2.89s         Wait for com.yaymedialabs.putong.enterprise to idle
> 2020-12-16 07:35:12:862 [Xcode] 
> 2020-12-16 07:35:27:868 [Xcode] 2020-12-16 15:35:27.865968+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe it is idling
> 2020-12-16 07:35:27:868 [Xcode] 
> 2020-12-16 07:35:27:869 [Xcode] 2020-12-16 15:35:27.866572+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe there are no animations

I would try to play with quiescence checks (`waitForIdleTimeout` capability/setting and/or `animationCoolOffTimeout` settings, both are measured in float seconds). It seems like the app is not idling enough, but it still makes sense to just limit the max idle wait duration rather than to cancel it completely by setting `waitForQuiescence` to `false`.

**Comment by yongjie6** [comment]
> seconds

Thanks，I try it

**Comment by yongjie6** [comment]
> > 12-16 07:35:12:861 [Xcode]     t =     2.89s         Wait for com.yaymedialabs.putong.enterprise to idle
> > 2020-12-16 07:35:12:862 [Xcode]
> > 2020-12-16 07:35:27:868 [Xcode] 2020-12-16 15:35:27.865968+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe it is idling
> > 2020-12-16 07:35:27:868 [Xcode]
> > 2020-12-16 07:35:27:869 [Xcode] 2020-12-16 15:35:27.866572+0800 WebDriverAgentRunner-Runner[735:122507] Quiescence checks are disabled for com.yaymedialabs.putong.enterprise application. Making it to believe there are no animations
> 
> I would try to play with quiescence checks (`waitForIdleTimeout` capability/setting and/or `animationCoolOffTimeout` setting, both are measured in float seconds). It seems like the app is not idling enough, but it still makes sense to just limit the max idle wait duration rather than to cancel it completely by setting `waitForQuiescence` to `false`.

Thanks，I try it

**Comment by RonCocoa** [comment]
it's working for me now, I am using 
Xcode 12.3,
Appium Desktop version 1.19.1,
Node js version 12.8.1.

**Comment by mykola-mokhnach** [comment]
Closed as third party issue

**Comment by muralitallapudi** [comment]
while going through this blog, I could not understand the fix. 
RonCocoa, Can you summarize the fix to this problem?.

**Comment by Jiurong-Yang** [comment]
We are seeing connections in this issue actually causing the Appium server to crash. This is the last log error message before our server mysteriously dies.
